### ISSUE TITLE
Improve error message when dashes are used in PROVIDES or REQUIRES entries

### ISSUE THREAD

**Issue by ghost** [haikuports/haikuporter#240]
I'm trying to package a command call diff-so-fancy.
The script is just a single perl script downloaded from github without external deps, hence a rather easy recipe.

I have in my recipe:
```
PROVIDES="
	cmd:diff-so-fancy
	"
```

Now this gives me the error
```
        recipe for diff_so_fancy-1.4.3 is still broken:
        Error: evaluating file /boot/home/proj/haikuports/app-misc/diff-so-fancy/work-1.4.3/port.recipe produced illegal value "cmd:diff-so-fancy" for key PROVIDES
        Error: dashes are not allowed in provides- or requires declarations
```

What should be done instead? escaping does not work. Is it ment to not be possible to package command names with dashes in them?

**Comment by augiedoggie** [comment]
https://github.com/haikuports/haikuports/wiki/HaikuPorter-HandlingRequirements#requiring-a-command

**Comment by pulkomandy** [comment]
Yes, but it never was a problem so far.

**Comment by pulkomandy** [comment]
yes, this is not a bug but an intentional choice. The provides/requires can't have dashes so the dash always separates the package name and the package version and there is no risk of confusion.

It seems unlikely we could have two different commands, one with underscores and the other with dashes. If that was the case, we would probably rename one to avoid the risof mixing them up? So this restriction is acceptable and solves a lot of other possible confusion cases.

What do you want to be documented? "replace dashes with underscores"?
